src/products/service/stock.service.ts

```diff
@@ -0,0 +1,33 @@
+import { BadRequestException, Injectable, NotFoundException } from "@nestjs/common";
+import { StockRepository } from "../repository/stock.repository";
+import { OrderItem } from "src/orders/dto/create-order.dto";
+import { Stock } from "../domain/stock.entity";
+
+export interface StockItem{
+    productId: string;
+    quantity: number;
+}
+@Injectable()
+export class StockService{
+
+    constructor(private readonly stockRepository: StockRepository){}
+
+    async get(orderItems: OrderItem[]): Promise<Stock[]>{
+        let stockQueries: Promise<Stock>[];
```

Comment:
**suggestion:** try to have the variable types unwrapped from the promise types (e.g.: just use `Stock[]`), it makes it easier to use later on

Reply:
Noted.
I actually removed that variable as I do not need it after the recent fix.